### ISSUE TITLE
Doesn't detect properly conditional expressions for `role`

### ISSUE THREAD

**Issue by feimosi** [evcohen/eslint-plugin-jsx-a11y#252]
I'm getting a false positive for the following example:
```jsx
const clickable = onClick !== noop;
return (
  <div
    onClick={ onClick }
    role={ clickable ? 'button' : null }
  >
     { /* */ }
  </div>
);
```

```
error  Static HTML elements with event handlers require a role 
jsx-a11y/no-static-element-interactions
```

**Comment by ljharb** [comment]
I would expect `onClick` to be *omitted* when it's not a role=button, not passed as a noop.

For a number of reasons, you'd probably want to do:
```jsx
if (clickable) {
  return <div onClick={onClick} role="button">{children}</div>;
}
return <div>{children}</div>;
```

i'm not sure how the linter would be capable of knowing that you consistently have a role=button when you have an onClick, nor how the linter would be able to know that "noop" is in fact a noop.

**Comment by feimosi** [comment]
Ok, actually you're right. I didn't think about it this way.
